projects/cfengine/string_fuzzer.c

```diff
@@ -0,0 +1,28 @@
+/* Copyright 2021 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <stdint.h>
+#include <string.h>
+#include <stdlib.h>
+#include <string_expressions.h>
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size){
+        char *new_str = (char *)malloc(size+1);
+        if (new_str == NULL){
+                return 0;
+        }
+        memcpy(new_str, data, size);
+        new_str[size] = '\0';
+        StringParseResult res = ParseStringExpression(new_str, 0, strlen(new_str));
```

Comment:
How/why did you choose `ParseStringExpression()`?

We've done some fuzzing before, manually, but then we've fuzzed the complete parser:

https://github.com/olehermanse/dotfiles/blob/master/cfengine/fuzz.sh

Reply:
Hitting the complete parser is a good goal, and doable as well. I will be glad to give it a go, but I would prefer this to be a separate endeavour once CFengine is up and running as a it requires a bit more effort than I have available at this moment.

Reply:
Sure, no problem, I was more interested in an answer as to why you chose this part.